README.md

```diff
@@ -6,7 +6,148 @@
 
 V3IO Frames is a high-speed server and client library for accessing time-series (TSDB), NoSQL, and streaming data in the [Iguazio Continuous Data Platform](https://www.iguazio.com).
 
-## Components
+## Documentation
+
+Frames currently supports 3 backends and basic CRUD functionality for each.
+
+Supported Backends:
+1. TSDB
+2. KV
+3. Stream
+4. CSV - for testing purposes
+
+
+All of frames operations are executed via the `client` object. To create a client object simply provide the Iguazio web-api endpoint and optional credentials.
+```python
+import v3io_frames as v3f
+client = v3f.Client('web-api:8081')
+```
+Next, for every operation we need to provide a `backend`, and a `table` parameters and optionally other function specific arguments.
+
+### Create
+Creates a new table for the wanted backend. Not all backends require a table to be created prior to ingestion. For example KV table will be created while ingesting new data, on the other hand since TSDB tables have mandatory fields we need to create a table before ingesting new data.  
+```python
+client.create(backend=<backend>, table=<table>, attrs=<backend_specefic_attributes>)
+```
+
+#### backend specific parameters
+##### TSDB
+* rate 
+* aggregates (optional)
+* aggregation-granularity (optional)
+
+For detailed info on these parameters please visit [TSDB](https://github.com/v3io/v3io-tsdb) docs.  
+Example:
+```python
+client.create('tsdb', '/mytable', attrs={'rate': '1/m'})
+```
+
+##### Stream
+* shards (optional)
+* retention_hours (optional)
+
+For detailed info on these parameters please visit Stream docs.  
+Example:
+```python
+client.create('stream', '/mystream', attrs={'shards': '6'})
+```
+
+### Write
+Writes a Dataframe into one of the supported backends.  
+Common write parameters:
+* dfs - list of Dataframes to write
+* index_cols=None (optional) - specify specific index columns, by default Dataframe's index columns will be used.
+* expression=' ' (optional)
+* condition=' ' (optional)
+* labels=None (optional)
+* max_in_message=0 (optional)
+* partition_keys=None (optional)
+
+Example:
+```python
+data = [['tom', 10], ['nick', 15], ['juli', 14]] 
+df = pd.DataFrame(data, columns = ['name', 'age'])
+df.set_index('name')
+client.write(backend='kv', table='mytable', dfs=df)
+```
+
+### Read
+Reads data from a backend.  
+Common read parameters:
+* query: string - Query in SQL format
+* iterator: bool - Return iterator of DataFrames or (if False) just one DataFrame
+* filter: string - Query filter (can't be used with query)
+* group_by: string - Query group by (can't be used with query)
+* columns: []str - List of columns to pass (can't be used with query)
+* limit: int - Maximal number of rows to return
+* data_format: string - Data format
+* row_layout: bool - Weather to use row layout (vs the default column layout)
+* max_in_message: int - Maximal number of rows per message
```

Comment:
As you can see in `Read` we have 2 kinda similar props:
1. limit - Basically **nobody** uses it, except CSV backend
2. max_in_message (default kv-256, stream-1024)- used in KV & Stream as the limit in messages between framesd and py pkg. Miki "implemented" his own limit logic without  using the Nginx limit option.

Also there is `max_in_message` in `client.Write(...)` which means maximum dfs that are being sent over to framesd at a time.. usually we will write only 1 df anyway.. 🤷‍♂ 

Might want to rethink how we expose these to the user and maybe change the logic we use when using these props. 
Anyway currently i only copied the existing properties to the README as is, so let's either remove them from docs and think how to improve it later, or leave it as is.

Reply:
cool for now. lets discuss what we want there for the long run as for read user will use iterator in case of large collections... I will hide these props to simplify the api

Reply:
I'm in favour of keeping it in the docs. just for reference and so that we will be aware of it. currently it's just us looking at it anyway